src/proteus/dswx_hls.py

```diff
@@ -3060,7 +3065,12 @@ def _create_ocean_mask(shapefile, margin_km, scratch_dir,
             temp_shapefile_filename = tempfile.NamedTemporaryFile(
                 dir=scratch_dir, suffix='.shp').name
             if temp_files_list is not None:
-                temp_files_list.append(temp_shapefile_filename)
+                for extension in ['.shp', '.prj', '.dbf', '.shx']:
+                    temp_file_test = temp_shapefile_filename.replace(
+                        '.shp', extension)
+                    if not os.path.isfile(temp_file_test):
+                        continue
```

Comment:
Unfortunately, this line is not going to work; the .prj, .dbf, and .shx files have not yet been created at this point in the code, and so they will not be added to the `temp_files_list`, and they are still leftover in the `scratch_dir`.

Instead, could this entire 7-line code block be moved to the end of the function (right before `return`)? I just tested this, and all temp files get removed this way.

Reply:
Good catch! Thank you, @nemo794 ! I remove that check.

Reply:
Great! Just tested again, the scratch_dir is empty now. Great!

Except now I'm thinking about this more, and I do really like the idea of doing an existence check before adding these "surprise" files to the list. Example scenario: if the caller of the function got this list back and tried to delete everything without first doing an existence check themself, they would get an error because the file does not exist.

In the current DSWx-HLS code, there is an existence check at the end of the code before deleting these files, so we will not get this error. But it's probably better practice to keep the existence check within this function, but perhaps add the files to the list at the end of this function?

Reply:
Just updated the code, @nemo794 . Please, let me know if you have any other comments or concerns. Thanks.